Eyaml gpg

smbambling opened the pull request:
Add the ability to install/configure hiera-eyaml-gpg. 
 * Configures needed hiera.yaml properties
 * Installs needed gems

Note:  Added puppetlabs/ruby dep as its the ruby-devel package is needed so it installs it

There is an oddity that if the pkcs7 keys for hiera-eyaml-gpg need created AND the hiera-eyaml-gpg gem is already installed it fails to create the keys and the run reports back failed.  To compensate for this I have a exec that will remove the hiera-eyaml-gpg ONLY if the pkcs7 keys have not been generated.  This lead to the potential for the hiera-eyaml-gpg gem to not be installed on a single run but will be install on the next puppet run.  I don't see this as being something that is run into often.


Example Usage

class { 'hiera':
  hierarchy => [
    'nodes/%{::clientcert}',
    'locations/%{::location}',
    'environments/%{::applicationtier}',
    'common',
  ],
  logger    => 'console',
  eyaml     => true,
  eyaml_gpg => true,
  eyaml_gpg_keygen => true,
  eyaml_gpg_recipients => 'sihil@example.com,gtmtech@example.com,tpoulton@example.com',
}

smbambling commented:
@hunner: I fixed the current merge conflicts. Any other requirements or suggestions to approve this request ?

igalic commented:
@smbambling rather than fixing mere conflicts, should should probably just [rebase](http://git-scm.com/book/en/Git-Branching-Rebasing) and [squash](http://www.git-scm.com/book/en/Git-Tools-Rewriting-History#Squashing-Commits)

smbambling commented:
@igalic I rebased my feature branch from master ( after fetching the upstream and merging to my local master )  there was still conflicts that needed to be fixed.  Though when I rebased I did not toss a -i arg to squash the commit.

Besides the rebase in the future any code suggestions ?

dmizelle commented:
Hey all,
I've went ahead and rebased off of hunner-hiera/master, and thats fixed the merge conflicts. Also, I've updated the README.md to include the new params.

What else can we do to help get this merged?

Cheers.

igalic commented:
re-rebase it, then bug @hunner.
